### PR TITLE
[CALCITE-4876] Converting RelNode to SQL with CalciteSqlDialect gets wrong result while EnumerableIntersect is followed by EnumerableLimit

### PR DESCRIPTION
**Pull request by ILuffZhe**
This PR adjusts the precedence among SqlSetOperators.

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 1]
```diff
@@ -1881,6 +1881,31 @@ private void checkHavingAliasSameAsColumn(boolean upperAlias) {
     relFn(relFn).ok(expectedSql);
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-4876">[CALCITE-4876]
+   * Converting RelNode to SQL with CalciteSqlDialect gets wrong result
+   * while EnumerableIntersect is followed by EnumerableLimit.</a>.
```
Comment (NobiGo) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:1887]: Please remove the redundant  '.'  after the EnumerableLimit.

**Review by NobiGo — CHANGES_REQUESTED** [review]

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java (hunk 1 above)
Comment (ILuffZhe) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java]: OK, I've removed it.

**Review by NobiGo — APPROVED** [review]

**Comment by NobiGo** [comment]
@ILuffZhe  Please rebase the commit info.

**Comment by ILuffZhe** [comment]
@NobiGo Done, thanks for your review.

**Pull request merged**